File: src/page.ts
import { Elysia } from "elysia";
import { html } from "@elysiajs/html";
import * as elements from "typed-html";
import { getActiveConfiguration } from "./utils/getData";

export default async function page() {
    function BaseHtml({ children }: elements.Children) {
        return `<!DOCTYPE html>
    <html lang="en">
      <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta http-equiv="X-UA-Compatible" content="ie=edge">
        <title>Lince</title>
        <!-- <link rel="stylesheet" href="style.css"> -->
      </head>
        ${children}
    </html>`;
    }

    const app = new Elysia()
        .use(html())
        .get("/", async () => {
            const data = await getActiveConfiguration();
            return data[0].configurationname;
        })
        .get("/based", ({ html }) => html(<BasebaseHtml))
        .listen(3000);
    console.log(`Serving at ${app.server?.hostname}:${app.server?.port}`);
}

page();

File: src/infrastructure/http/handlers/section/page.rs
use axum::response::{Html, IntoResponse};

pub async fn page_handler() -> impl IntoResponse {
    Html(
        br#"
    <!doctype html>
    <html lang="en">
        <head>
            <meta charset="UTF-8" />
            <meta name="viewport" content="width=device-width, initial-scale=1.0" />
            <meta http-equiv="X-UA-Compatible" content="ie=edge" />
            <title>Lince</title>
            <script src="https://unpkg.com/htmx.org@2.0.4"></script>
            <style>
                body {
                    background-color: #000000;
                    color: #ffffff;
                }
            </style>
        </head>
        <body
            id="body"
            hx-trigger="load"
            hx-get="/section/body"
            hx-swap="outerHTML"
        ></body>
    </html>
    "#,
    )
}
